operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -0,0 +1,125 @@
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.utils.K8sUtils.deployKeycloak;
+import static org.keycloak.operator.utils.K8sUtils.getDefaultKeycloakDeployment;
+import static org.keycloak.operator.utils.K8sUtils.waitForKeycloakToBeReady;
+
+@QuarkusTest
+public class KeycloakDeploymentE2EIT extends ClusterOperatorTest {
+    @Test
+    public void testBasicKeycloakDeploymentAndDeletion() {
+        try {
+            // CR
+            Log.info("Creating new Keycloak CR example");
+            var kc = getDefaultKeycloakDeployment(k8sclient);
+            var deploymentName = kc.getMetadata().getName();
+            deployKeycloak(k8sclient, kc, true);
+
+            // Check Operator has deployed Keycloak
+            Log.info("Checking Operator has deployed Keycloak deployment");
+            assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get()).isNotNull();
+
+            // Check Keycloak has correct replicas
+            Log.info("Checking Keycloak pod has ready replicas == 1");
+            assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getReadyReplicas()).isEqualTo(1);
+
+            // Delete CR
+            Log.info("Deleting Keycloak CR and watching cleanup");
+            k8sclient.resources(Keycloak.class).delete(kc);
+            Awaitility.await()
+                    .untilAsserted(() -> assertThat(k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get()).isNull());
+        } catch (Exception e) {
+            savePodLogs();
+            throw e;
+        }
+    }
+
+    @Test
+    public void testCRFields() {
+        try {
+            var kc = getDefaultKeycloakDeployment(k8sclient);
+            var deploymentName = kc.getMetadata().getName();
+            deployKeycloak(k8sclient, kc, true);
+
+            kc.getSpec().setImage("quay.io/keycloak/non-existing-keycloak");
+            kc.getSpec().getServerConfiguration().put("KC_DB_PASSWORD", "Ay Caramba!");
+            deployKeycloak(k8sclient, kc, false);
+
+            Awaitility.await()
+                    .during(Duration.ofSeconds(15)) // check if the Deployment is stable
+                    .untilAsserted(() -> {
+                        var c = k8sclient.apps().deployments().inNamespace(namespace).withName(deploymentName).get()
+                                .getSpec().getTemplate().getSpec().getContainers().get(0);
+                        assertThat(c.getImage()).isEqualTo("quay.io/keycloak/non-existing-keycloak");
+                        assertThat(c.getEnv().stream()
+                                .anyMatch(e -> e.getName().equals("KC_DB_PASSWORD") && e.getValue().equals("Ay Caramba!")))
```

Comment:
jajajajaj Amazing Password !!!!! :)

Reply:
Should we have these hardcoded values as constants ? "KC_DB_PASSWORD"

Reply:
☝️  as a reminder for the future :-)